### PR TITLE
feat(monitors): Pass environment through

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Supports specifying the checkins environment

### CODE REVIEW

**Review by rjo100 — APPROVED** [review]
thank you for this!!

**Review by olksdr — APPROVED** [review]

File: relay-monitors/src/lib.rs [hunk 1]
```diff
@@ -113,6 +113,10 @@ struct CheckIn {
     /// Status of this check-in. Defaults to `"unknown"`.
     status: CheckInStatus,
 
+    /// The environment to associate the check-in with
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    environment: Option<String>,
```
Comment (iker-barriocanal) [inline relay-monitors/src/lib.rs:118]: Let's add some length limits to this string, like [here](https://github.com/getsentry/relay/blob/ab2a3dca7a3daaecfcdf097d898c766fd6b69476/relay-general/src/protocol/event.rs#L381). Monitor slugs may contain non-ASCII characters, if this isn't the case for `environment` consider also adding [`allow_chars`](https://github.com/getsentry/relay/blob/ab2a3dca7a3daaecfcdf097d898c766fd6b69476/relay-general/src/protocol/event.rs#L368).

**Review by iker-barriocanal — APPROVED** [review]

File: relay-monitors/src/lib.rs (hunk 1 above)
Comment (evanpurkhiser) [reply relay-monitors/src/lib.rs:118]: Sounds good. Thanks for linking this

Comment (evanpurkhiser) [reply relay-monitors/src/lib.rs:118]: Will merge and update this

Comment (evanpurkhiser) [reply relay-monitors/src/lib.rs:118]: Actually, it looks like events uses a completely separate serialization system? I'm not familiar with `metastructure`

Comment (evanpurkhiser) [reply relay-monitors/src/lib.rs:118]: Going to merge this as is and revisit validation

**Pull request merged**